clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -409,15 +434,18 @@ public void testNonExpiringMetadata() throws Exception {
         assertTrue("Unused topic expired when expiry disabled", metadata.containsTopic("topic4"));
     }
 
+    private void clearBackgroundError() {
+        backgroundError.set(null);
+    }
+
     private Thread asyncFetch(final String topic, final long maxWaitMs) {
         Thread thread = new Thread() {
             public void run() {
-                while (metadata.fetch().partitionsForTopic(topic).isEmpty()) {
-                    try {
+                try {
```

Comment:
Hmm.. The old logic would let us continue fetching in the case of a timeout. Do you think that was not intentional?

Reply:
Even if we continued fetching, we would have failed the test at the end. `tearDown()` checks if we saw any background errors and fails the test if we did, so I thought that this change should be reasonable. Let me know if you think otherwise.